six/modules/c++/cphd/include/cphd/PVPBlock.h

```diff
@@ -0,0 +1,409 @@
+         *  will be written to
+         */
+        void read(const Pvp& pvp, sys::ubyte* output) const;
+
+        //! Equality operators
+        bool operator==(const PVPSet& other) const
+        {
+            if(!(txTime == other.txTime && txPos == other.txPos &&
+                    txVel == other.txVel && rcvTime == other.rcvTime &&
+                    rcvPos == other.rcvPos && rcvVel == other.rcvVel &&
+                    srpPos == other.srpPos && aFDOP == other.aFDOP &&
+                    aFRR1 == other.aFRR1 && aFRR2 == other.aFRR2 &&
+                    fx1 == other.fx1 && fx2 == other.fx2 &&
+                    toa1 == other.toa1 && toa2 == other.toa2 &&
+                    tdTropoSRP == other.tdTropoSRP && sc0 == other.sc0 &&
+                    scss == other.scss &&
+                    ampSF == other.ampSF && fxN1 == other.fxN1 &&
+                    fxN2 == other.fxN2 && toaE1 == other.toaE1 &&
+                    toaE2 == other.toaE2 && tdIonoSRP == other.tdIonoSRP &&
+                    signal == other.signal))
+            {
+                return false;
+            }
+                if (addedPVP.size() != other.addedPVP.size())
+                {
+                    return false;
+                }
+                for (auto it = addedPVP.begin(); it != addedPVP.end(); ++it)
+                {
+                    if(it->second.str() != other.addedPVP.find(it->first)->second.str())
+                    {
+                        return false;
+                    }
+                }
+                return true;
+        }
+        bool operator!=(const PVPSet& other) const
+        {
+            return !((*this) == other);
+        }
+
+        //! Required Parameters
+        double txTime;
+        Vector3 txPos;
+        Vector3 txVel;
+        double rcvTime;
+        Vector3 rcvPos;
+        Vector3 rcvVel;
+        Vector3 srpPos;
+        double aFDOP;
+        double aFRR1;
+        double aFRR2;
+        double fx1;
+        double fx2;
+        double toa1;
+        double toa2;
+        double tdTropoSRP;
+        double sc0;
+        double scss;
+
+        //! (Optional) Parameters
+        mem::ScopedCopyablePtr<double> ampSF;
+        mem::ScopedCopyablePtr<double> fxN1;
+        mem::ScopedCopyablePtr<double> fxN2;
+        mem::ScopedCopyablePtr<double> toaE1;
+        mem::ScopedCopyablePtr<double> toaE2;
+        mem::ScopedCopyablePtr<double> tdIonoSRP;
+        mem::ScopedCopyablePtr<double> signal;
+
+        //! (Optional) Additional parameters
+        std::map<std::string,six::Parameter> addedPVP;
+    };
+    friend std::ostream& operator<< (std::ostream& os, const PVPSet& p);
+
+private:
+    /*
+     *  Struct Template Specialization for getAddedPVP function
+     *  mimics function template specialization
+     */
+    template<typename T>
+    struct AddedPVP
+    {
+        T getAddedPVP(const six::Parameter& val, size_t channel, size_t set, const std::string& name) const
```

Comment:
I don't see why you need the other parameters here

Reply:
Initially thought these functions could do more. Removed them now.